### ISSUE TITLE
Inventory Description Design Decision

### ISSUE THREAD

**Issue by chrismeyersfsu** [chrismeyersfsu/provision_docker#3]
Currently, the `provision_docker` role exposes an interface much like an Ansible cloud module. The role variable `provision_docker_inventory` is used to express the desired set of docker containers to bring up. One alternative to this interface that I have toyed with is to use "dummy inventory" entries to express the set of docker containers that you want to bring up/operate on. For example:

```
# inventory file
[group1]
host1 image="chrismeyers/ubuntu12.04"
host2

[group2]
host1
host2
host3
```

The "dummy inventory" method allows the user to add more cleanly add hostvars. I say more cleanly because with the existing cloud-module-like interface you can still hadd hostvars. However, you have to do so in code using `set_fact`, which is not as clean.

With the cloud-module-like interface I've already added and exposed a `groups` parameter so the user doesn't have to add the host to a group in code. I don't want to go down the path of adding more one-off features like this when we can get it for free with the dummy-inventory method.

So the items up for discussion becomes, should we ...
- transition to a dummy-inventory interface?
- support both the dummy-inventory and cloud-module-like interface?

If we stick with the cloud-module-like interface, what features should we add ...
- parameter to specify arbitrary facts to attach to docker container/hosts
- other?

**Comment by jakemcdermott** [comment]
@chrismeyersfsu 
One advantage to going with the inventory interface is that it would make it easier to model a role's test inventory (host variables and all) after parts of a real inventory or set of inventories being used in a project.

Having a workflow where you can drop into a role's directory and make changes to it against a dynamically created, container-based test environment that you've specced out with a realistic static inventory file and then jumping back up to the project level to use it in playbooks against inventories that point to real machines is powerful.

**Comment by chrismeyersfsu** [comment]
I tend to agree with you. The programmer in me wants the cloud-like interface, but the inventory interface just seems so elegant (ansible and docker-compose like).

**Comment by chrismeyersfsu** [comment]
@jakemcdermott I'm having trouble with one aspect of the inventory interface.
https://github.com/chrismeyersfsu/provision_docker/blob/feature-inv_specification/tasks/inc_inventory_iface.yml#L21

I want to update facts for the "dummy" hosts. Specifically, the ip address. However, I am unable because `set_fact` operates on `localhost`

**Comment by chrismeyersfsu** [comment]
Problem solved, `delegate_to` + `delegate_facts` ,fixed #6
